### PR TITLE
T337002 api

### PR DESCRIPTION
**Pull request by emkarcinos**
I've moved all API classes into a separate folder,
as I felt like grouping modules improves readability.
APIs themselves had storage logic which I extracted and put
in another folder. I was originally going with an interface
to the storage to allow for other storage methods than
log entries, but the code was too tightly coupled with it,
so I've left that for another day. Added dependency injection
for all services and used ServiceOptions for config vars.

### CODE REVIEW

File: includes/Api/ApiThank.php [hunk 1]
```diff
@@ -77,53 +98,12 @@ protected function markResultSuccess( $recipientName ) {
 		] );
 	}
 
-	/**
-	 * This checks the log_search data.
-	 *
-	 * @param User $thanker The user sending the thanks.
-	 * @param string $uniqueId The identifier for the thanks.
-	 * @return bool Whether thanks has already been sent
-	 */
```
Comment (mszabo-wikia) [inline includes/Api/ApiThank.php:86]: We should preserve these comments

File: includes/Storage/LogStorage.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	 * @param ILoadBalancer $lb
+	 * @param ActorNormalization $actorNormalization
+	 */
+	public function __construct( ILoadBalancer $lb, ActorNormalization $actorNormalization ) {
+		$this->lb = $lb;
+		$this->actorNormalization = $actorNormalization;
+	}
+
+	public function thank( User $user, User $recipient, $uniqueId ) {
+		if ( !RequestContext::getMain()->getConfig()->get( 'ThanksLogging' ) ) {
```
Comment (mszabo-wikia) [inline includes/Storage/LogStorage.php]: This config value could be injected via ServiceOptions

File: includes/Storage/LogStorage.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			// TODO: This should be done in a separate hook handler
+			Hooks::updateCheckUserData( $recentChange );
+		}
+	}
+
+	/**
+	 * This checks the log_search data.
+	 */
+	public function haveThanked( User $thanker, string $uniqueId ): bool {
+		// TODO: Figure out why it's not getting the data from a replica
```
Comment (mszabo-wikia) [inline includes/Storage/LogStorage.php:66]: Does it try to read data that may have been written in the current request?

File: includes/Storage/LogStorage.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+			)
+			->caller( __METHOD__ )
+			->fetchRow();
+	}
+
+	/**
+	 * @throws InvalidLogType
+	 * @throws LogDeleted
+	 */
+	public function getLogEntryFromId( Config $config, int $id ): ?DatabaseLogEntry {
```
Comment (mszabo-wikia) [inline includes/Storage/LogStorage.php]: If we inject config vars via ServiceOptions then this also need not take a Config argument.

File: tests/phpunit/ApiCoreThankIntegrationTest.php [hunk 5]
```diff
@@ -74,7 +74,7 @@ public function setUp(): void {
 	}
 
 	public function testRequestWithoutToken() {
-		$this->expectApiErrorCode( 'missingparam' );
+		$this->expectExceptionMessage( 'The "token" parameter must be set' );
```
Comment (mszabo-wikia) [inline tests/phpunit/ApiCoreThankIntegrationTest.php]: are we changing the API contract here?

**Comment by emkarcinos** [comment]
https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Thanks/+/921397

**Pull request closed without merging**